### PR TITLE
Music asset manager

### PR DESCRIPTION
**Pull request by Cavonstavant**
# This pull request adds audio support to the projet.

## Misc modifications
- `AssetLoader`  was updated to allow multiple assets type mgmt
    - currently loading: `textures`, `audio` and `fonts`
- All assets are loaded inside the `AssetLoader` and can be accessed by using the associated getter type

## New features
A new class `audio::AudioManager` implements two ways of interracting with audio within the project.
Audio is managed in two ways. Buffered and non-buffered.
### Buffered
This type is proper to use with small, short audio files like sound effects. It loads the file into memory using an `sf::SoundBuffer` and can be played with latency.
This type is refered to a SFX all along the code

### Non-buffered
This type is proper to use with large, long audio files like background music. It uses an `sf::Music` that is based on an `sf::SoundStream`.
> Unlike audio buffers (see [sf::SoundBuffer](https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1SoundBuffer.php)), audio streams are never completely loaded in memory.
> Instead, the audio data is acquired continuously while the stream is playing. This behavior allows to play a sound with no loading delay, and keeps the memory consumption very low.

This type is refered to a BGM all along the code

### CODE REVIEW

**Review by aurelenc — CHANGES_REQUESTED** [review]
- Replace tabs by 4 spaces
- Remove copyrighted songs

**Review by GuyomT — APPROVED** [review]
not building but code is clear

LGTM

File: .gdb_history [hunk 1]
```diff
@@ -60,3 +60,5 @@ p *this
 q
```
Comment (Bootoyka) [inline .gdb_history]: Please ensure that you don't push .gdb_history

**Review by Bootoyka — APPROVED** [review]
Still looking for the correlation between the pull request's name and the content but lgtm

**Review by aurelenc — APPROVED** [review]

**Pull request merged**